apis/python/src/tiledbsoma/_sparse_nd_array.py

```diff
@@ -292,14 +292,18 @@ def __init__(self, sr: clib.SOMAReader, shape: NTuple):
         self.sr = sr
         self.shape = shape
 
-    def coos(self) -> SparseCOOTensorReadIter:
+    def coos(self, shape: Optional[NTuple] = None) -> SparseCOOTensorReadIter:
         """
         Returns an iterator of Arrow SparseCOOTensor.
 
+        Args:
+            shape:
+                Optionally, a NTuple that overrides the default capacity.
+
         Lifecycle:
             Experimental.
         """
-        return SparseCOOTensorReadIter(self.sr, self.shape)
+        return SparseCOOTensorReadIter(self.sr, shape or self.shape)
```

Comment:
very simple and very elegant! :) 

let's add a check that (if shape is not None) `len(shape) == len(self.shape)` and raise `ValueError` if not -- with a unit-test `pytest.raises(ValueError)` for this as well